myproject_sql/tests/tests.py

```diff
@@ -0,0 +1,127 @@
+    config.include("..models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture(scope="function")
+def populated_db(request, sqlengine):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(Entry(title="Day 15", body="This is a test entry, James is being awesome.", creation_date=datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(Entry).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Call a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_model_gets_added(new_session):
+    assert len(new_session.query(Entry).all()) == 0
+    model = Entry(title="Bob", body="Test")
+    new_session.add(model)
+    new_session.flush()
+    assert len(new_session.query(Entry).all()) == 1
+
+
+def test_home_view(dummy_request, new_session):
+    """Test entries on the home view."""
+    from ..views.views import my_view
+    new_session.add(Entry(title='the title', body='the body', creation_date=datetime.utcnow()))
+    new_session.flush()
+    info = my_view(dummy_request)
+    # import pdb; pdb.set_trace()
+    assert info['entries'][0].title == 'the title'
+
+
+def test_create_view(dummy_request, new_session):
+    """Test entries on the home view."""
+    from ..views.views import create
+    info = create(dummy_request)
+    assert info == {}
+
+
+# def test_create_view(dummy_request, new_session):
+#     """Test entries on the home view."""
+#     from ..views.views import create
+#     info = create(dummy_request)
+
+
+# ----------Functional--------------------
+
+@pytest.fixture()
+def testapp(sqlengine):
+    from myproject_sql import main
+    app = main({}, **DB_SETTINGS)
+    from webtest import TestApp
+    return TestApp(app)
+
+
+def test_layout_main(testapp, populated_db):
+    response = testapp.get('/', status=200)
+    assert b'Day 15' in response.body
+
+
+def test_layout_single_entry(testapp, populated_db):
+    response = testapp.get('/journal/1', status=200)
+    assert b'This is a test entry' in response.body
+
+
+def test_layout_edit_entry(testapp, populated_db):
+    response = testapp.get('/journal/1/edit-entry', status=200)
+    assert b'This is a test entry' in response.body
```

Comment:
for a layout test like this, I'd urge you to use `response.html` instead, so you can make more literate assertions about the HTML structure of your page.  There's probably a number of things about that layout you could assert, no?  More than just that a given entry body is present.  For example, it might be nice to assert not only that the body text of your entry is present in the rendered page, but that it is the content of a `<textarea>` tag.  After all, if I render it into the page as just plain text, not in a form input, it's not going to do me any good in trying to edit it.